wagtail_streamfield_migration_toolkit/test/tests/test_old_list.py

```diff
@@ -0,0 +1,262 @@
+from django.test import TestCase
+
+from .. import models
+from wagtail_streamfield_migration_toolkit.utils import apply_changes_to_raw_data
+from wagtail_streamfield_migration_toolkit.operations import (
+    RenameStreamChildrenOperation,
+    RenameStructChildrenOperation,
+    ListChildrenToStructBlockOperation,
+)
+
+
+class OldListFormatNestedStreamTestCase(TestCase):
+    """Tests involving changes to ListBlocks in the old format with StreamBlock children"""
+
+    @classmethod
+    def setUpTestData(cls):
+        raw_data = [
+            {"type": "char1", "id": "0001", "value": "Char Block 1"},
+            {
+                "type": "nestedlist_stream",
+                "id": "0002",
+                "value": [
+                    [
+                        {"type": "char1", "id": "0003", "value": "Char Block 1"},
+                        {"type": "char2", "id": "0004", "value": "Char Block 2"},
+                        {"type": "char1", "id": "0005", "value": "Char Block 1"},
+                    ],
+                    [
+                        {"type": "char1", "id": "0006", "value": "Char Block 1"},
+                    ],
+                ],
+            },
+            {
+                "type": "nestedlist_stream",
+                "id": "0007",
+                "value": [
+                    [
+                        {"type": "char1", "id": "0008", "value": "Char Block 1"},
+                    ]
+                ],
+            },
+        ]
+        cls.raw_data = raw_data
+
+    def test_list_converted_to_new_format_in_recursion(self):
+        """Test whether all ListBlock children have converted formats during the recursion.
+
+        This tests the changes done in the recursion process only, so the operation used isn't
+        important. We will use a rename operation for now.
+
+        Check whether each ListBlock child has attributes id, value, type and type is item.
+        Check whether rename operation was done successfully.
+        """
+
+        altered_raw_data = apply_changes_to_raw_data(
+            self.raw_data,
+            "nestedlist_stream.item",
+            RenameStreamChildrenOperation(old_name="char1", new_name="renamed1"),
+            streamfield=models.SampleModel.content,
+        )
+
+        for listitem in altered_raw_data[1]["value"]:
+            self.assertIsInstance(listitem, dict)
+            self.assertIn("type", listitem)
+            self.assertIn("value", listitem)
+            self.assertEqual(listitem["type"], "item")
+
+        for listitem in altered_raw_data[2]["value"]:
+            self.assertIsInstance(listitem, dict)
+            self.assertIn("type", listitem)
+            self.assertIn("value", listitem)
+            self.assertEqual(listitem["type"], "item")
+
+        altered_block_path_indices = [
```

Comment:
Could you add an explanatory comment here as to what these represent? Is it just the indices of blocks where the type should have changed?